src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/CommandItemViewModel.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
+using Microsoft.CmdPal.UI.ViewModels.Models;
+
+namespace Microsoft.CmdPal.UI.ViewModels;
+
+public partial class CommandItemViewModel : ObservableObject
+{
+    private readonly ExtensionObject<ICommandItem> _commandItemModel = new(null);
+
+    // private bool _initialized;
+    public string Name { get; private set; } = string.Empty;
+
+    public string Title { get; private set; } = string.Empty;
+
+    public string Subtitle { get; private set; } = string.Empty;
+
+    public string IconUri { get; private set; } = string.Empty;
+
+    public ExtensionObject<ICommand> Command { get; } = new(null);
```

Comment:
Note: this change breaks the UI that was navigating before, as it used to just be an `ICommand` here: https://github.com/zadjii-msft/PowerToys/pull/175/files#diff-455f6ec74df03e77f910b618b63881b69c35dd679d820aad4e0648a85075ce10L23 (guess it's too different to keep it tracked through the rename from ActionBarContextItemViewModel that it was based off of).

The in `ShellPage.xaml.cs:Receive(PerformCommandMessage message)` the command type is checked against both `IListPage` and `IInvokableCommand` in order to execute the command.